lib/jobs/secure-erase-drive.js

```diff
@@ -0,0 +1,410 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = secureEraseJobFactory;
+di.annotate(secureEraseJobFactory, new di.Provide('Job.Drive.SecureErase'));
+    di.annotate(secureEraseJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'Logger',
+        'Promise',
+        'Assert',
+        'Util',
+        '_',
+        'JobUtils.Commands',
+        'JobUtils.CatalogSearchHelpers'
+    )
+);
+
+function secureEraseJobFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    _,
+    CommandUtil,
+    catalogSearch
+) {
+    var logger = Logger.initialize(secureEraseJobFactory);
+    /**
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function SecureEraseJob(options, context, taskId) {
+        SecureEraseJob.super_.call(this, logger, options, context, taskId);
+        assert.string(this.context.target);
+        assert.arrayOfObject(this.options.eraseSettings, 'eraseSettings');
+
+        this.commandUtil = new CommandUtil(this.context.target);
+        this.nodeId = this.context.target;
+
+        this.eraseSettings = {};
+        this.commands = [];
+        this.hasSentCommands = false;
+
+        this.driveConst = Object.freeze({
+            hdparm: {
+                protocol: ['SATADOM', 'SATA'],
+                args: ['secure-erase', 'secure-erase-enhanced']
+            },
+            sg_sanitize: { //jshint ignore:line
+                protocol: ['SAS'],
+                args: ['block', 'crypto', 'fail']
+            },
+            sg_format: { //jshint ignore:line
+                protocol: ['SAS'],
+                args: ['0', '1']
+            },
+            scrub: {
+                protocol: undefined,
+                args: ['nnsa', 'dod', 'fillzero', 'random', 'random2', 'fillff', 'gutmann',
+                    'schneier', 'fastold', 'pfitzner7', 'pfitzner33', 'usarmy', 'old', 'fastold',
+                    'custom=string']
```

Comment:
custom=string, string should be a character input by customer, like 0x00, 0xff, 0x81